utils/merkletrie/filesystem/node_test.go

```diff
@@ -0,0 +1,114 @@
+package filesystem
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	. "gopkg.in/check.v1"
+	"gopkg.in/src-d/go-billy.v2"
+	"gopkg.in/src-d/go-billy.v2/memfs"
+	"gopkg.in/src-d/go-git.v4/utils/merkletrie"
+	"gopkg.in/src-d/go-git.v4/utils/merkletrie/noder"
+)
+
+func Test(t *testing.T) { TestingT(t) }
+
+type NoderSuite struct{}
+
+var _ = Suite(&NoderSuite{})
+
+func (s *NoderSuite) TestDiff(c *C) {
```

Comment:
I think this way of testing `node` will make debugging harder than if we do unit testing of its public methods instead: it would be easier to detect errors with the unit tests than having to trace back difftree's results mismatches in the tests to see what errors in `node` caused them.

I would write unit tests for `node` and maybe left a cople of these as integrations tests.